### ISSUE TITLE
It's not possible put a null gateway in a network bridge

### ISSUE THREAD

**Issue by maykopetersen** [bpg/terraform-provider-proxmox#1089]
**Describe the bug**
Proxmox force us to put a Gateway during its installation. This is a not good policy, because there are teams that has a Network Management with only IP and Subnet mask without connections with other networks.

How did I think to erase Gateway from the `vmbr0` (default bridge in Proxmox)? Using `import` and `resource proxmox_virtual_environment_network_linux_bridge` blocks together, but there was an error during the `apply`:

```sh
$ tofu apply "plano"
proxmox_virtual_environment_network_linux_bridge.vmbr0: Importing... [id=node01:vmbr0]
proxmox_virtual_environment_network_linux_bridge.vmbr0: Import complete [id=node01:vmbr0]
module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0: Creating...
proxmox_virtual_environment_network_linux_bridge.vmbr0: Modifying... [id=node01:vmbr0]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to proxmox_virtual_environment_network_linux_bridge.vmbr0, provider
│ "provider[\"registry.opentofu.org/bpg/proxmox\"].bpg_nodes" produced an unexpected new value: .gateway: was null, but
│ now cty.StringVal("10.10.10.254").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Error creating Linux Bridge interface
│
│   with module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0,
│   on modules/nodes/node.tf line 93, in resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0":
│   93: resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
│
│ Could not create Linux Bridge, unexpected error: failed to create network interface "vmbr0" for node "node01":
│ received an HTTP 400 response - Reason: Parameter verification failed. (iface: interface already exists)
```

**To Reproduce**

1. Create the `main.tf` file:
```hcl
terraform {
  required_providers {
    proxmox = {
      source  = "bpg/proxmox"
      version = "0.43.2"
    }
  }
}

provider "proxmox" {
  alias    = "bpg_nodes"
  endpoint = "https://${lookup(var.nodes_ips, var.node_target)}:8006/"
  insecure = true
  ssh {
    agent = true
  }
}
```
2. Create the `variables.tf`file:
```hcl
variable "nodes_ips" {
  type        = map(any)
  default = {
    node01 = "10.10.10.10"
  }
}

variable "nodes_ips_cidr" {
  type = number
  default = 24
}

variable "node_target" {
  type        = string
}
```
3. Create the `import.tf` file:
```hcl
import {
  to = proxmox_virtual_environment_network_linux_bridge.vmbr0
  id = "${var.node_target}:vmbr0"
}

resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
  provider = proxmox.bpg_nodes
  node_name = var.node_target
  name      = "vmbr0"
  address = "${lookup(var.nodes_ips, var.node_target)}/${var.nodes_ips_cidr}"
  ports = ["eno1"]
  autostart = true
  comment = "Managed by Terraform"
  gateway = null
}
```
4. Validate and plan ok:
```sh
$ tofu validate .
Success! The configuration is valid.

$ tofu plan -out=plano -var="node_target"="node01"
proxmox_virtual_environment_network_linux_bridge.vmbr0: Preparing import... [id=node01:vmbr0]
proxmox_virtual_environment_network_linux_bridge.vmbr0: Refreshing state... [id=node01:vmbr0]

OpenTofu used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create
  ~ update in-place

OpenTofu will perform the following actions:

  # proxmox_virtual_environment_network_linux_bridge.vmbr0 will be updated in-place
  # (imported from "node01:vmbr0")
  ~ resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
        address    = "10.10.10.10/24"
        autostart  = true
      + comment    = "Managed by Terraform"
      - gateway    = "10.10.10.254" -> null
        id         = "node01:vmbr0"
        name       = "vmbr0"
        node_name  = "node01"
        ports      = [
            "eno1",
        ]
      ~ vlan_aware = false -> (known after apply)
    }

  # module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0 will be created
  + resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
      + autostart  = true
      + comment    = "Managed by Terraform"
      + id         = (known after apply)
      + name       = "vmbr0"
      + node_name  = "node01"
      + vlan_aware = (known after apply)
    }

Plan: 1 to import, 1 to add, 1 to change, 0 to destroy.

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Saved the plan to: plano

To perform exactly these actions, run the following command to apply:
    tofu apply "plano"
```
5. You only see the error during the `apply` (I'll put it again):
```sh
$ tofu apply "plano"
proxmox_virtual_environment_network_linux_bridge.vmbr0: Importing... [id=node01:vmbr0]
proxmox_virtual_environment_network_linux_bridge.vmbr0: Import complete [id=node01:vmbr0]
module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0: Creating...
proxmox_virtual_environment_network_linux_bridge.vmbr0: Modifying... [id=node01:vmbr0]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to proxmox_virtual_environment_network_linux_bridge.vmbr0, provider
│ "provider[\"registry.opentofu.org/bpg/proxmox\"].bpg_nodes" produced an unexpected new value: .gateway: was null, but
│ now cty.StringVal("10.10.10.254").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Error creating Linux Bridge interface
│
│   with module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0,
│   on modules/nodes/node.tf line 93, in resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0":
│   93: resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
│
│ Could not create Linux Bridge, unexpected error: failed to create network interface "vmbr0" for node "node01":
│ received an HTTP 400 response - Reason: Parameter verification failed. (iface: interface already exists)
╵
```

**Expected behavior**
Remove Gateway address from `vmbr1` and import this configuration to my `terraform.tfstate`.

**Screenshots**
A very curious thing is the `resource` put the `comment`, but it doesn't remove the gateway:
![BPG-vmbr1-screenshot](https://github.com/bpg/terraform-provider-proxmox/assets/127991237/982f130b-aa55-4d21-8afd-a8a6465ba4a0)

**Additional context**
I think that or there is a validate in Gateway or it is a mandatory variable. If gateway is a optional variable or validate can be `null` value , the problem is resolved :)

- Single Proxmox
- Provider version (ideally it should be the latest version): 8.0.3
- Terraform version: OpenTofu v1.6.2
- OS (where you run Terraform from): Debian GNU/Linux 10

**Comment by bpg** [comment]
Hey @maykopetersen 👋🏼 

https://github.com/bpg/terraform-provider-proxmox/pull/1095 may fix it. But I don't really have a way to test it, my lab is in a subnet 🤷🏼

**Comment by maykopetersen** [comment]
> Hey @maykopetersen 👋🏼
> 
> #1095 may fix it. But I don't really have a way to test it, my lab is in a subnet 🤷🏼

Thank you very much! I saw the code and you added the follow:

```go
	if !plan.Gateway.Equal(state.Gateway) && plan.Gateway.ValueString() == "" {
		toDelete = append(toDelete, "gateway")
		body.Gateway = nil
	}
```

I just can to do a test in friday, but I believe that this will resolve :)

**Comment by maykopetersen** [comment]
Is this fix in a `0.48.1` version already?
1. I updated my `terraform` block:
```hcl
terraform {
  required_providers {
    proxmox = {
      source  = "bpg/proxmox"
      version = "0.48.1"
    }
  }
}
```
2. Runned `tofu init -upgrade` command.
3. Modified my `resource`:
```hcl
# file: module/node.tf
resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
  node_name = var.node_target
  name      = "vmbr0"
  address = var.address
  ports = ["eno1"]
  autostart = true
  comment = "Managed by Terraform"
  gateway = null
#  gateway = ""
  vlan_aware = false
}
```
4. Plan:
```sh
$ tofu plan -out=plano -var="node_target"="node01"
...
OpenTofu will perform the following actions:

  # module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0 will be updated in-place
  ~ resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0" {
      - gateway    = "10.10.10.254" -> null
```
5. And I had the same error in `apply`:
```sh
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0, provider
│ "provider[\"registry.opentofu.org/bpg/proxmox\"].bpg_nodes" produced an unexpected new value: .gateway: was null, but
│ now cty.StringVal("10.10.10.254").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

I saw you put:
```go
	if !plan.Gateway.Equal(state.Gateway) && plan.Gateway.ValueString() == "" {
		toDelete = append(toDelete, "gateway")
		body.Gateway = nil
	}
```

If I configure the `gateway = ""` (rather than `null`) in plan step I have an erro:
```sh
╷
│ Error: Invalid IP String Value
│
│   with module.node_config.proxmox_virtual_environment_network_linux_bridge.vmbr0,
│   on modules/nodes/node.tf line 40, in resource "proxmox_virtual_environment_network_linux_bridge" "vmbr0":
│   40:   gateway = ""
│
│ An unexpected error occurred while converting a string value that was expected to be IPv4/IPv6.
│
│ Path: gateway
│ Given Value:
╵
```

**Comment by bpg** [comment]
No, it will be in `0.48.2`, not merged yet.
